homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,554 @@
+            )
+            sensors.append(
+                OmnilogicSensor(
+                    coordinator,
+                    "salt_level",
+                    "Salt Level",
+                    backyard,
+                    bow,
+                    "none",
+                    "mdi:gauge",
+                    PERCENT_UNITS,
+                    bow["Chlorinator"],
+                )
+            )
+
+            if "CSAD" in bow.keys():
+                if bow["CSAD"]["systemId"] != "0":
+                    _LOGGER.info("Setting up ph sensor for %s.", bow.get("Name"))
+                    sensors.append(
+                        OmnilogicSensor(
+                            coordinator,
+                            "csad_ph",
+                            "CSAD",
+                            backyard,
+                            bow,
+                            "none",
+                            "mdi:gauge",
+                            PERCENT_UNITS,
+                            bow["CSAD"],
+                        )
+                    )
+
+                    _LOGGER.info("Setting up orp sensor for %s.", bow.get("Name"))
+                    sensors.append(
+                        OmnilogicSensor(
+                            coordinator,
+                            "csad_orp",
+                            "CSAD",
+                            backyard,
+                            bow,
+                            "none",
+                            "mdi:gauge",
+                            PERCENT_UNITS,
+                            bow["CSAD"],
+                        )
+                    )
+
+    async_add_entities(sensors, update_before_add=True)
+
+
+class OmnilogicSensor(Entity):
+    """Defines an Omnilogic sensor entity."""
+
+    def __init__(
+        self,
+        coordinator,
+        kind,
+        name,
+        backyard,
+        bow,
+        device_class,
+        icon,
+        unit,
+        sensordata,
+    ):
+        """Initialize Entities."""
+
+        if bow != {}:
+            sensorname = (
+                backyard["BackyardName"].replace(" ", "_")
+                + "_"
+                + bow["Name"].replace(" ", "_")
+                + "_"
+                + kind
+            )
+        else:
+            sensorname = backyard["BackyardName"].replace(" ", "_") + "_" + kind
+
+        self._kind = kind
+        self._name = None
+        self.entity_id = ENTITY_ID_FORMAT.format(sensorname)
+        self._backyard = backyard
+        self._backyard_name = backyard["BackyardName"]
+        self._state = None
+        self._unit_type = backyard["Unit-of-Measurement"]
+        self._device_class = device_class
+        self._icon = icon
+        self._bow = bow
+        self._unit = None
+        self.coordinator = coordinator
+        self.bow = bow
+        self.sensordata = sensordata
+        self.attrs = {}
+        self.attrs["MspSystemId"] = backyard["systemId"]
```

Comment:
Also, just initialize to `{"MspSystemId"]: backyard["systemId"]}`. No need for two steps

Reply:
Addressed.